pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -643,37 +731,127 @@ func (sip *streamIngestionProcessor) bufferCheckpoint(event partitionEvent) erro
 	return nil
 }
 
+// Write a batch of MVCC range keys into the SST batcher, and returns
+// the current size of all buffered range keys.
+func (r *rangeKeyBatcher) write(rangeKeys mvccRangeKeyValues) (int, error) {
+	if len(rangeKeys) == 0 {
+		return 0, nil
+	}
+	// Lazily create the SST writer when we really need it.
+	if r.rangeKeySSTWriter == nil {
+		r.rangeKeySSTWriter = r.rangeKeySSTWriterMaker()
+	}
+
+	for _, rangeKeyVal := range rangeKeys {
+		mvccValue, err := storage.DecodeMVCCValue(rangeKeyVal.Value)
+		if err != nil {
+			return 0, err
+		}
+		if err = r.rangeKeySSTWriter.PutMVCCRangeKey(rangeKeyVal.RangeKey, mvccValue); err != nil {
```

Comment:
Looks like you might be able to use `PutRawMVCCRangeKey` now to avoid the decode.